File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/domain/IsShowOnHoldUseCase.java
package com.glia.widgets.view.floatingvisitorvideoview.domain;

import com.glia.widgets.helper.rx.Schedulers;

import io.reactivex.Single;

public class IsShowOnHoldUseCase {
    private final Schedulers schedulers;

    public IsShowOnHoldUseCase(Schedulers schedulers) {
        this.schedulers = schedulers;
    }

    public Single<Boolean> execute(boolean isOnHold) {
        return Single.just(isOnHold)
                .subscribeOn(schedulers.getComputationScheduler())
                .observeOn(schedulers.getMainScheduler());
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/domain/IsShowOnHoldUseCase.kt
package com.glia.widgets.view.floatingvisitorvideoview.domain

import com.glia.widgets.helper.rx.Schedulers
import io.reactivex.Single

internal class IsShowOnHoldUseCase(private val schedulers: Schedulers) {
    operator fun invoke(isOnHold: Boolean): Single<Boolean> {
        return Single.just(isOnHold)
            .subscribeOn(schedulers.computationScheduler)
            .observeOn(schedulers.mainScheduler)
    }
}
